OAK-Marketplace-Emotion-Recognition/emotion_analyzer.py

```diff
@@ -0,0 +1,245 @@
+            alpha_mask = cur_bar.alpha_mask
+            result_statistic = overlay_image(result_statistic, cur_bar.bar, start_x, start_y, alpha_mask)
+            start_x = np.clip(start_x - result_bar_size - result_bar_padding, 0, result_statistic.shape[1])
+        if save:
+            cv2.imwrite(save_path, result_statistic)
+        return Image.fromarray(result_statistic[:, :, ::-1])
+
+
+class EmotionBar:
+    def __init__(
+        self,
+        bar_size: int = 80,
+        emotion: str = 'happy',
+        progress: float = 0.0,
+        emotion_size: tuple = (25, 25),
+        emotion_path: str = 'images/emoji/happy.png',
+        **kwargs
+    ):
+        self._progress = progress
+        self.bar = np.zeros((bar_size, bar_size, 3), dtype=np.uint8)
+        self.alpha_mask = np.ones((bar_size, bar_size, 3), dtype=np.uint8)
+        self.emotion_image = np.array(Image.open(emotion_path).resize(emotion_size))
+        self.bar_size = bar_size
+        self.emotion_size = emotion_size
+        self.emotion = emotion
+        self._init_params(**kwargs)
+
+    def _init_params(self, **kwargs):
+        self.background_color = kwargs.get("background_color", (28, 28, 30))
+        self.start_angle = kwargs.get("start_angle", -220)
+        self.progress_color = kwargs.get("progress_color", (145, 82, 225))
+        self.thickness = kwargs.get("thickness", 10)
+        self.end_angle = kwargs.get("end_angle", 40)
+        self.radius = self.bar_size // 2
+        self.center = (self.bar_size // 2, self.bar_size // 2)
+        self.axes = (self.radius, self.radius)
+
+    @property
+    def progress(self):
+        return self._progress
+
+    @progress.setter
+    def progress(self, value):
+        self._progress = value
+
+    def draw(self, progress: float = 0.0):
+        self.progress = progress
+        diff = (self.end_angle - self.start_angle) * progress
+        self.bar = cv2.ellipse(
+            self.bar, center=self.center, axes=self.axes, angle=0, startAngle=self.start_angle,
+            endAngle=self.end_angle, color=self.background_color, thickness=-1,
+        )
+        self.bar = cv2.ellipse(
+            self.bar, center=self.center, axes=self.axes, angle=0, startAngle=self.start_angle,
+            endAngle=self.start_angle + diff, color=self.progress_color, thickness=-1,
+        )
+        self.bar = cv2.ellipse(
+            self.bar, center=self.center, axes=(self.radius - self.thickness, self.radius - self.thickness),
+            angle=0, startAngle=self.start_angle - 1, endAngle=self.end_angle + 1,
+            color=(0, 0, 0), thickness=-1,
+        )
+        alpha_mask = self.emotion_image[:, :, 3] / 255.0
+        overlay_emotion = self.emotion_image[:, :, :3][:,:,::-1]
+        self.bar = overlay_image(
+            self.bar, overlay_emotion, int(self.center[0] - self.emotion_image.shape[1] / 2),
+            int(self.center[1] - self.emotion_image.shape[0] / 2), alpha_mask,
+        )
```

Comment:
Please, add comments on what's being done here? 

Also, please, think about how to get rid of the overlay_image() function.